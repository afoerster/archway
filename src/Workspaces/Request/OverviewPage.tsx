import * as React from 'react';
import {Col, Form, Input, Checkbox} from 'antd';
import {withFormik, InjectedFormikProps, FormikFormProps} from 'formik';
import FieldLabel from '../../Common/FieldLabel';
import {Request} from './model';

const OverviewPage = ({ 
  handleChange, 
  handleSubmit, 
  values: { 
    name, 
    summary,
    description,
    phi_data,
    pii_data,
    pci_data
  } }: InjectedFormikProps<FormikFormProps, Request>) => (
  <Col span={12} offset={6}>
    <Form
      layout="vertical"
      onSubmit={handleSubmit}>
      <Form.Item
        label={<FieldLabel>Workspace Name</FieldLabel>}>
        <Input
          size="large"
          name="name"
          placeholder="Loan Modification Group (LMG)"
          value={name}
          onChange={handleChange} />
      </Form.Item>
      <Form.Item
        label={<FieldLabel>Summary</FieldLabel>}>
        <Input
          size="large"
          name="summary"
          placeholder="LMG's place to evaluate modification algorithms"
          value={summary}
          onChange={handleChange} />
      </Form.Item>
      <Form.Item
        label={<FieldLabel>Purpose for workspace</FieldLabel>}>
        <Input.TextArea
          name="description"
          placeholder="(Use this area to fully describe to reviewers and future users why this workspace is needed.)"
          rows={5}
          value={description}
          onChange={handleChange} />
      </Form.Item>
      <Form.Item
        label={<FieldLabel>The data in this workspace may contain</FieldLabel>}>
        <Checkbox.Group onChange={handleChange}>
          <Checkbox name="compliance" value="phi_data">PHI</Checkbox>
          <Checkbox name="compliance" value="pci_data">PCI</Checkbox>
          <Checkbox name="compliance" value="pii_data">PII</Checkbox>
        </Checkbox.Group>
      </Form.Item>
    </Form>
  </Col>
);

export default withFormik({
  handleSubmit: (payload, { props }: any) => {
    props.setRequest(payload);
  }
})(OverviewPage);;