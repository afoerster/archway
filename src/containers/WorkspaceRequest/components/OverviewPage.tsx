import { Checkbox, Col, Form, Input } from 'antd';
import * as React from 'react';
import { FieldLabel } from '../../../components';
import { RequestInput } from '../../../models/RequestInput';

interface Props {
  request: RequestInput,
  setRequest: (request: RequestInput | boolean) => void;
}

export default class OverviewPage extends React.Component<Props> {
  public onChangeValue = (key: string) => (e: any) => {
    const { request, setRequest } = this.props;
    setRequest({
      ...request,
      [key]: e.target.value,
    });
  }

  public onChangeCheckbox = (key: string) => (e: any) => {
    const { request, setRequest } = this.props;
    setRequest({
      ...request,
      compliance: {
        ...request.compliance,
        [key]: e.target.checked,
      },
    });
  }

  public render() {
    const { request } = this.props;

    return (
      <Col span={12} offset={6}>
        <Form layout="vertical">
          <Form.Item
            label={<FieldLabel>Workspace Name</FieldLabel>}>
            <Input
              size="large"
              name="name"
              placeholder="Loan Modification Group (LMG)"
              value={request.name}
              onChange={this.onChangeValue('name')}
            />
          </Form.Item>
          <Form.Item
            label={<FieldLabel>Summary</FieldLabel>}>
            <Input
              size="large"
              name="summary"
              placeholder="LMG's place to evaluate modification algorithms"
              value={request.summary}
              onChange={this.onChangeValue('summary')}
            />
          </Form.Item>
          <Form.Item
            label={<FieldLabel>Purpose for workspace</FieldLabel>}>
            <Input.TextArea
              name="description"
              /* tslint:disable:max-line-length */
              placeholder="(Use this area to fully describe to reviewers and future users why this workspace is needed.)"
              rows={5}
              value={request.description}
              onChange={this.onChangeValue('description')}
            />
          </Form.Item>
          <Form.Item
            label={<FieldLabel>The data in this workspace may contain</FieldLabel>}>
            <Checkbox
              name="phi_data"
              checked={request.compliance.phi_data}
              onChange={this.onChangeCheckbox('phi_data')}
            >
              PHI
            </Checkbox>
            <Checkbox
              name="pci_data"
              checked={request.compliance.pci_data}
              onChange={this.onChangeCheckbox('pci_data')}
            >
              PCI
            </Checkbox>
            <Checkbox
              name="pii_data"
              checked={request.compliance.pii_data}
              onChange={this.onChangeCheckbox('pii_data')}
            >
              PII
            </Checkbox>
          </Form.Item>
        </Form>
      </Col>
    );
  }
}
