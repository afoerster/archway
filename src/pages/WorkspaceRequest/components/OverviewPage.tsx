import { Checkbox, Col, Form, Input } from 'antd';
import { CheckboxChangeEvent } from 'antd/lib/checkbox';
import { FormikBag, InjectedFormikProps, withFormik } from 'formik';
import * as React from 'react';
import * as Yup from 'yup';
import { FieldLabel } from '../../../components';
import { RequestInput } from '../../../types/RequestInput';

interface Props {
  setRequest: (request: RequestInput) => void;
}

const RequestSchema = Yup.object().shape({
  name: Yup.string().required(),
  summary: Yup.string().required(),
  description: Yup.string().required(),
});

class OverviewPage extends React.PureComponent<InjectedFormikProps<Props, RequestInput>> {

    constructor(props: InjectedFormikProps<Props, RequestInput>) {
    super(props);

    this.updateComplianceItem = this.updateComplianceItem.bind(this);
    this.onChangeWrapper = this.onChangeWrapper.bind(this);
  }

    public render() {
    const {
      handleSubmit,
      errors,
      touched,
      values: {
        name,
        summary,
        description,
        compliance: {
          phi_data,
          pii_data,
          pci_data,
        },
      } } = this.props;

    return (
      <Col span={12} offset={6}>
        <Form
          layout="vertical"
          onSubmit={handleSubmit}>
          <Form.Item
            validateStatus={errors.name && touched.name ? 'error' : 'success'}
            label={<FieldLabel>Workspace Name</FieldLabel>}>
            <Input
              size="large"
              name="name"
              placeholder="Loan Modification Group (LMG)"
              value={name}
              onChange={this.onChangeWrapper} />
          </Form.Item>
          <Form.Item
            validateStatus={errors.summary && touched.summary ? 'error' : 'success'}
            label={<FieldLabel>Summary</FieldLabel>}>
            <Input
              size="large"
              name="summary"
              placeholder="LMG's place to evaluate modification algorithms"
              value={summary}
              onChange={this.onChangeWrapper} />
          </Form.Item>
          <Form.Item
            validateStatus={errors.description && touched.description ? 'error' : 'success'}
            label={<FieldLabel>Purpose for workspace</FieldLabel>}>
            <Input.TextArea
              name="description"
              /* tslint:disable:max-line-length */
              placeholder="(Use this area to fully describe to reviewers and future users why this workspace is needed.)"
              rows={5}
              value={description}
              onChange={this.onChangeWrapper} />
          </Form.Item>
          <Form.Item
            label={<FieldLabel>The data in this workspace may contain</FieldLabel>}>
            <Checkbox name="phi_data" checked={phi_data} onChange={this.updateComplianceItem}>PHI</Checkbox>
            <Checkbox name="pci_data" checked={pci_data} onChange={this.updateComplianceItem}>PCI</Checkbox>
            <Checkbox name="pii_data" checked={pii_data} onChange={this.updateComplianceItem}>PII</Checkbox>
          </Form.Item>
        </Form>
      </Col>
    );
  }

    private updateComplianceItem(e: CheckboxChangeEvent) {
    this.props.setFieldValue(`compliance.${e.target.name!}`, e.target.checked);
    this.props.submitForm();
  }

    private onChangeWrapper(e: any) {
    this.props.handleChange(e);
    this.props.submitForm();
  }
}

export default withFormik({
  validationSchema: RequestSchema,
  handleSubmit: (payload: RequestInput, { props }: FormikBag<Props, RequestInput>) => {
    props.setRequest(payload);
  },
  mapPropsToValues: () => ({
    name: '',
    summary: '',
    description: '',
    compliance: {
      phi_data: false,
      pii_data: false,
      pci_data: false,
    },
  }),
})(OverviewPage);
