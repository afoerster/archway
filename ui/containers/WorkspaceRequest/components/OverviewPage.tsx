import { Col, Form, Input } from 'antd';
import * as React from 'react';
import { FieldLabel } from '../../../components';
import { RequestInput } from '../../../models/RequestInput';
import { FormComponentProps } from 'antd/lib/form/Form';

interface Props extends FormComponentProps {
  request: RequestInput;
  setRequest: (request: RequestInput | boolean) => void;
}

class OverviewPage extends React.Component<Props> {
  public onChangeValue = (key: string) => (e: any) => {
    const { request, setRequest } = this.props;
    setRequest({
      ...request,
      [key]: e.target.value.trim().length ? e.target.value : '',
    });
  };

  public render() {
    const { request, form } = this.props;
    const { getFieldDecorator, isFieldTouched, getFieldError } = form;
    const nameError = isFieldTouched('name') && getFieldError('name');
    const summaryError = isFieldTouched('summary') && getFieldError('summary');
    const descriptionError = isFieldTouched('description') && getFieldError('description');

    return (
      <Col span={12} offset={6}>
        <Form layout="vertical">
          <FieldLabel>Workspace Name</FieldLabel>
          <Form.Item style={{ marginTop: 8 }} validateStatus={nameError ? 'error' : ''}>
            {getFieldDecorator('name', {
              rules: [{ required: true, message: 'Please enter your workspace name!' }],
            })(
              <Input
                size="large"
                name="name"
                placeholder="Loan Modification Group (LMG)"
                value={request.name}
                onChange={this.onChangeValue('name')}
              />
            )}
          </Form.Item>
          <FieldLabel>Summary</FieldLabel>
          <Form.Item style={{ marginTop: 8 }} validateStatus={summaryError ? 'error' : ''}>
            {getFieldDecorator('summary', {
              rules: [{ required: true, message: `Please enter your summary!` }],
            })(
              <Input
                size="large"
                name="summary"
                placeholder="LMG's place to evaluate modification algorithms"
                value={request.summary}
                onChange={this.onChangeValue('summary')}
              />
            )}
          </Form.Item>
          <FieldLabel>Purpose for workspace</FieldLabel>
          <Form.Item style={{ marginTop: 8 }} validateStatus={descriptionError ? 'error' : ''}>
            {getFieldDecorator('description', {
              rules: [{ required: true, message: `Please enter your description!` }],
            })(
              <Input.TextArea
                name="description"
                /* tslint:disable:max-line-length */
                placeholder="(Use this area to fully describe to reviewers and future users why this workspace is needed.)"
                rows={5}
                value={request.description}
                onChange={this.onChangeValue('description')}
              />
            )}
          </Form.Item>
        </Form>
      </Col>
    );
  }
}

export default Form.create<Props>()(OverviewPage);
