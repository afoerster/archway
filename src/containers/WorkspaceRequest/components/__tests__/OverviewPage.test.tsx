import * as React from 'react';
import { shallow } from 'enzyme';

import OverviewPage from '../OverviewPage';

describe('OverviewPage', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<OverviewPage setRequest={() => null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
