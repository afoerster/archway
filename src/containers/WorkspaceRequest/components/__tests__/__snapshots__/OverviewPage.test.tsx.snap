// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewPage renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithFormik(OverviewPage)
    setRequest={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "enableReinitialize": false,
      "handleSubmit": [Function],
      "initialValues": Object {
        "compliance": Object {
          "pci_data": false,
          "phi_data": false,
          "pii_data": false,
        },
        "description": "",
        "name": "",
        "summary": "",
      },
      "isInitialValid": false,
      "onSubmit": [Function],
      "render": [Function],
      "setRequest": [Function],
      "validate": undefined,
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enableReinitialize": false,
        "handleSubmit": [Function],
        "initialValues": Object {
          "compliance": Object {
            "pci_data": false,
            "phi_data": false,
            "pii_data": false,
          },
          "description": "",
          "name": "",
          "summary": "",
        },
        "isInitialValid": false,
        "onSubmit": [Function],
        "render": [Function],
        "setRequest": [Function],
        "validate": undefined,
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
